hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/HFileWriterImpl.java

```diff
@@ -555,16 +563,33 @@ private void writeInlineBlocks(boolean closing) throws IOException {
   private void doCacheOnWrite(long offset) {
     cacheConf.getBlockCache().ifPresent(cache -> {
       HFileBlock cacheFormatBlock = blockWriter.getBlockForCaching(cacheConf);
+      BlockCacheKey key = buildBlockCacheKey(offset, cacheFormatBlock);
+      if (!shouldCacheBlock(cache, key)) {
```

Comment:
I see that shouldCacheBlock internally makes a call to isHotData (key), which would go all the way to fetch the file metadata and decide if the file is hot or not. Would this be called per block? If so, would it impact the performance adversely?

Reply:
> which would go all the way to fetch the file metadata and decide if the file is hot or not.
Yes, it will look up the Configuration which corresponds to the store to which this file belongs to from the map. However, it won't read any file-related metadata, since we are already passing the maxTimestamp here.

> Would this be called per block?
Yes, it is called for every block.

> would it impact the performance adversely?
Yes, I think it will add some performance overhead.

> Can we somehow restrict these traversals to once per file instead of once per block.
I couldn't think of any better solution at the moment.